zathura/database-sqlite.c

```diff
@@ -242,6 +244,37 @@ sqlite_db_init(ZathuraSQLDatabase* db, const char* path)
     }
   }
 
+  /* check existing tables for correct column types */
+  ret1 = check_column_type(session, "fileinfo", "first_page_column", "TEXT", &res1);
+
+  if (ret1 == true && res1 == false) {
+    girara_debug("old database table layout detected; updating ...");
+
+    /* prepare transaction */
+    char tx_begin[] = "BEGIN TRANSACTION;"
+                      "ALTER TABLE fileinfo RENAME TO tmp;";
+    char tx_end[] = "INSERT INTO fileinfo SELECT * FROM tmp;"
+                    "DROP TABLE tmp;"
+                    "COMMIT;";
+
+    /* calculate requred buffer size */
+    size_t tx_buffer_size = strlen(tx_begin);
+    tx_buffer_size += strlen(SQL_FILEINFO_INIT);
+    tx_buffer_size += strlen(tx_end);
+    ++tx_buffer_size;
+
+    /* assemble transaction */
+    char transaction[tx_buffer_size];
```

Comment:
`char transaction[sizeof(tx_begin) + sizeof(SQL_FILEINFO_INIT) + sizeof(tx_end)] = { '\0' }` and remove the `bzero`.